x/bank/keeper/msg_server.go

```diff
@@ -181,3 +181,39 @@ func (k msgServer) SetSendEnabled(goCtx context.Context, msg *types.MsgSetSendEn
 
 	return &types.MsgSetSendEnabledResponse{}, nil
 }
+
+func (k msgServer) Burn(goCtx context.Context, msg *types.MsgBurn) (*types.MsgBurnResponse, error) {
+	var (
+		from []byte
+		err  error
+	)
+
+	var coins sdk.Coins
+	for _, coin := range msg.Amount {
+		coins = append(coins, sdk.NewCoin(coin.Denom, coin.Amount))
+	}
```

Comment:
@tac0turtle I believe there is a method inside Coins.Add which handles deduplication https://pkg.go.dev/github.com/cosmos/cosmos-sdk/types#Coins.Add otherwise currently are we sure that msg.Amount will have unique coins?